modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -262,13 +262,13 @@ public Object visit(PropertyIsLessThanOrEqualTo filter, Object extraData) {
     @Override
     public Object visit(PropertyIsLike filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
-        String expr = convert(filter.accept(this, null), String.class);
+        String expr = convert(filter.getExpression().accept(this, null), String.class);
```

Comment:
As pointed out by Andrea, no function is listed in the filter capabilities, so `filter.getExpression()` should never return an instance of `Function`. In any case, I added a check to make sure the filter expression is a `PropertyName` instance, as Mongo's $regex operator cannot deal with anything other than a field name.